### PR TITLE
Hayes branch

### PR DESCRIPTION
**Pull request by mjhayes412**
Rearranged how build.py works which decreased load time by 50.6% simply by pushing each new object in an array (instead of saving) and afterward using the bulk_create command to save all at once.

Also converted data into json files to take advantage of Django's loaddata functionality which is  76% faster than the original build.py 

json files are saved in pokemon_v2/fixtures per model for easier management.

### CODE REVIEW

**Comment by phalt** [comment]
@mjhayes412 This branch is actually too big to accurately review. Could you split it into a few less PRs with are easier to look at? I'd make the json fixtures a separate PR, and the code changes a different one.

**Comment by nimaje** [comment]
@mjhayes412 some files are readed twice (should be 10 after my counting for #154)
And I am against putting autogenerated files under versioncontrol.

`build_generic` could do someting like this:
```python
class2models = collections.defalutdict(list)
for data in daten:
    for model in data_to_model(data):
        class2models[model.__class__].append(model)
for model_class, models in class2models.items():
    model_class.objects.bulk_create(models)
```
this would solve the multiple reading

I don't think you want/should use `'w+'` as open mode (in `dump_one`, `dump_all`)
In `dump_all` you use a `with` statement and `close` (L1655)

**Comment by mjhayes412** [comment]
@nimaje Good points.  I knew about the twice read files for some and I thought about you when I came across them.  I figured I could come back to those 10 cases are refine it but I wanted to keep moving forward. I hope I was going in the direction you were intending.  It made the page look much cleaner.

Ah, I thought I took out that `w+` first.  I ran into a issue where it wouldn't create the file if it didn't exist but I realized I had a `/` at the start of my file name.

Why are you against putting autogenerated files under source control?  It's basically the same thing as providing the csv data files, except the json files load much, much faster.  

@phalt Thanks for the consideration! I'll try and break them up but I may try to improve them with nimaje's suggestions. Also, I should've ran through all the checks before submitting.  I don't have ci/circleci connected and when I ran `make test` I had some errors pop up.

image

**Comment by phalt** [comment]
@mjhayes412 for that error just run `make install` to install the new dependencies.

@nimaje raises a good point, we could auto generate them on the server / CI and then we can just save some space, making the project smaller. Considering we'd need to update it each time the data changes as well...

File: Makefile [hunk 1]
```diff
@@ -1,27 +1,27 @@
 install:
-	pip install -r requirements.txt
+    pip install -r requirements.txt
```
Comment (nimaje) [inline Makefile:2]: `make` works only with hard tabs
simple demo:

``` sh
$ echo "all:\n    @echo 'test'" > Makefile && make all
Makefile:2: *** missing separator.  Stop.
$ echo "all:\n\t@echo 'test'" > Makefile && make all                                                                                                                              :(
test
```

**Pull request closed without merging**